go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -107,9 +107,9 @@ func (a *PodAllocator) Init() error {
 }
 
 // getActiveNetworkForNamespace returns the active network for the given pod's namespace
-// and is a wrapper around util.GetActiveNetworkForNamespace
+// and is a wrapper around GetActiveNetworkForNamespace
 func (a *PodAllocator) getActiveNetworkForPod(pod *corev1.Pod) (util.NetInfo, error) {
-	activeNetwork, err := util.GetActiveNetworkForNamespace(pod.Namespace, a.nadLister)
+	activeNetwork, err := a.nadController.GetActiveNetworkForNamespace(pod.Namespace)
```

Comment:
I am going to flag some use cases where, if we had per-namespace NAD information in NetInfo, we probably wouldn't need access to the nad controller to query this information. This is one of them.

We would lose the protection of actually querying UDNs, true.

Just for awareness.

Reply:
it looks like this was intentional for some reason:
```
func (a *PodAllocator) GetNetworkRole(pod *corev1.Pod) (string, error) {
	if !util.IsNetworkSegmentationSupportEnabled() {
		// if user defined network segmentation is not enabled
		// then we know pod's primary network is "default" and
		// pod's secondary network is NOT its primary network
		if a.netInfo.IsDefault() {
			return types.NetworkRolePrimary, nil
		}
		return types.NetworkRoleSecondary, nil
	}
	activeNetwork, err := a.getActiveNetworkForPod(pod)
	if err != nil {
		return "", err
	}
	if activeNetwork.GetNetworkName() == a.netInfo.GetNetworkName() {
		return types.NetworkRolePrimary, nil
	}
	if a.netInfo.IsDefault() {
		// if default network was not the primary network,
		// then when UDN is turned on, default network is the
		// infrastructure-locked network forthis pod
		return types.NetworkRoleInfrastructure, nil
	}
	return types.NetworkRoleSecondary, nil
}
```

It is comparing the network name of the network found for the pod and the netinfo held in the allocator. @qinqon maybe you wrote the original code and can explain why it is like this? We are wondering why you cant just use the NetInfo on the allocator itself to know whether or not this is a primary or secondary role?

Reply:
@trozet `activeNetwork` is the pod's namespace primary network so this checks if the pod allocator network is the pod's namespace primary network (default or UDN) and terturn Primary according to that.

Reply:
Makes sense.

However this can be equally achieved if we could query the controller NetInfo for the namespaces served by the network and checking if the one we are concerned about is among them.

Anyway, I will follow up with this on a different PR maybe.

Reply:
Make sense too, thanks @jcaamano